src/main/java/at/hannibal2/skyhanni/features/combat/BroodmotherFeatures.kt

```diff
@@ -0,0 +1,157 @@
+
+        // ignore if the stage temporarily reverts to Imminent after the Broodmother's death
+        if (currentStage == StageEntry.IMMINENT && lastStage == StageEntry.ALIVE) return
+
+        if (currentStage == StageEntry.ALIVE) {
+            onBroodmotherSpawn()
+            return
+        }
+
+        val timeUntilSpawn = currentStage?.minutes?.minutes
+        broodmotherSpawnTime = SimpleTimeMark.now() + timeUntilSpawn!!
+
+        if (currentStage == StageEntry.IMMINENT && config.imminentWarning) {
+            playImminentWarning()
+            return
+        }
+
+        if (config.stages.contains(currentStage)) {
+            if (currentStage == StageEntry.SLAIN) {
+                onBroodmotherSlain()
+            } else {
+                val pluralize = StringUtils.pluralize(timeUntilSpawn.toInt(DurationUnit.MINUTES), "minute")
+                ChatUtils.chat(
+                    "Broodmother: $lastStage §e-> $currentStage§e. §b${timeUntilSpawn.inWholeMinutes} $pluralize §euntil it spawns!"
+                )
+            }
+        }
+    }
+
+    private fun onServerJoin() {
+        if (config.stageOnJoin && currentStage != StageEntry.ALIVE) {
+            ChatUtils.chat("The Broodmother's current stage in this server is $currentStage§r.")
+        }
+    }
+
+    private fun onBroodmotherSpawn() {
+        broodmotherSpawnTime = SimpleTimeMark.farPast()
+        if (config.spawnAlert || currentStage == StageEntry.ALIVE) {
+            ChatUtils.clickToActionOrDisable(
+                "The Broodmother has spawned!",
+                config::spawnAlert,
+                actionName = "warp to the Spider Mound",
+                action = { HypixelCommands.warp("nest") },
+            )
+        }
+        if (config.spawnAlert) {
+            val alertSound = SoundUtils.createSound(spawnAlertConfig.alertSound, spawnAlertConfig.pitch)
+            SoundUtils.repeatSound(100, spawnAlertConfig.repeatSound, alertSound)
+            LorenzUtils.sendTitle(spawnAlertConfig.text.replace("&", "§"), 3.seconds)
+        }
+    }
+
+    private fun playImminentWarning() {
+        SoundUtils.repeatSound(100, 2, SoundUtils.createSound("note.pling", 0.5f))
+        ChatUtils.chat("The Broodmother is §4Imminent§e! It will spawn in §b60 seconds§e!")
+    }
+
+    private fun onBroodmotherSlain() {
+        broodmotherSpawnTime = SimpleTimeMark.now() + 10.minutes
+        if (!(config.hideSlainWhenNearby && LorenzUtils.skyBlockArea == "Spider Mound")) {
+            ChatUtils.chat("The Broodmother was killed!")
+        }
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        broodmotherSpawnTime = SimpleTimeMark.farPast()
+        lastStage = null
+        currentStage = null
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!inSpidersDen()) return
+        if (!config.countdown) return
+        if (broodmotherSpawnTime.isInPast()) {
```

Comment:
please cache the string to show, only update it 10ish times a second instead of on each frame.